test/rules/omit_type_for_local_variables.dart

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N omit_type_for_local_variables`
+
+Map<int, List<Person>> badGroupByZip(Iterable<Person> people) {
+  Map<int, List<Person>> peopleByZip = <int, List<Person>>{}; // LINT
+
+  for (Person person in people) { // LINT
+    peopleByZip.putIfAbsent(person.zip, () => <Person>[]);
+    peopleByZip[person.zip].add(person);
+  }
+
+  return peopleByZip;
+}
+
+Map<int, List<Person>> goodGroupByZip(Iterable<Person> people) {
+  var peopleByZip = <int, List<Person>>{}; // OK
```

Comment:
Fixed

This is a test, it is ok to do not follow a coding style rules in order to test the rule you want to test.

But, I only tested the rule with var, so now is one with var and one with final to test better, thanks!